cli/providers/aws/elb/wait.go

```diff
@@ -9,43 +9,70 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/pchico83/i2kit/cli/schemas/service"
 )
 
-//Wait waits for the instances in a ELB to be registered
-func Wait(name string, config *aws.Config, log *logger.Logger) error {
+//Wait waits for the instances in a ELB to be healthy
+func Wait(s *service.Service, name string, config *aws.Config, log *logger.Logger) error {
+	registeredInstances, err := waitForRegistered(s, name, config, log)
+	if err != nil {
+		return err
+	}
+	log.Printf("Waiting for %s to be healthy in the '%s' load balancer...", registeredInstances, name)
+	for _, instanceID := range registeredInstances {
+		if err := waitForHealthy(name, instanceID, config, log); err != nil {
+			return err
+		}
+	}
+	log.Printf("%v are healthy in the '%s' load balancer.", registeredInstances, name)
+	return nil
+}
+
+func waitForRegistered(s *service.Service, name string, config *aws.Config, log *logger.Logger) ([]string, error) {
 	log.Printf("Waiting for instances to be registered in the '%s' load balancer...", name)
+	retries := 0
+	registeredInstances := []string{}
+	for retries < 30 {
+		registeredInstances = Instances(name, config)
+		log.Printf("%v instances registered in the '%s' load balancer.", registeredInstances, name)
+		if len(registeredInstances) == s.Replicas {
+			break
+		}
+		time.Sleep(10 * time.Second)
+	}
+	if retries == 20 {
```

Comment:
the max in the loop is 30, but you're checking for 20 here. Typo? I think it would be better to just return on line 39, and assume that if the loop exited, it's because there was an error.

Reply:
Hahahaha, I remember this discussion with you several times.
That it is not very academic, since exit conditions wouldn't be trivial.
We should follow the same rules every where. 
My proposal is:
 - Success conditions must be explicit in the loop variable.
 - Error can be returned in the middle of the loop.

In any case, this is not very relevant to me, I am ok following any other approach.

Reply:
Let's follow your proposal. Loop condition just handles the number of iterations.